src/lminiz.c

```diff
@@ -214,6 +219,126 @@ static int lmz_writer_finalize(lua_State *L) {
   return 1;
 }
 
+static int lmz_deflator_init(lua_State* L) {
+  int level = luaL_optint(L, 1, MZ_DEFAULT_COMPRESSION);
+  if (level < MZ_DEFAULT_COMPRESSION || level > MZ_BEST_COMPRESSION) {
+      luaL_error(L, "Compression level must be between -1 and 9");
+  }
+  lmz_stream_t* stream = lua_newuserdata(L, sizeof(*stream));
+  mz_streamp miniz_stream = &(stream->stream);
+  luaL_getmetatable(L, "miniz_deflator");
+  lua_setmetatable(L, -2);
+  memset(miniz_stream, 0, sizeof(*miniz_stream));
+  int status = mz_deflateInit(miniz_stream, level);
+  if (status != MZ_OK) {
+    const char* msg = mz_error(status);
+    if (msg) {
+      luaL_error(L, "Problem initializing stream: %s", msg);
+    } else {
+      luaL_error(L, "Problem initializing stream");
+    }
+  }
+  stream->mode = 0;
+  return 1;
+}
+
+static int lmz_inflator_init(lua_State* L) {
+  lmz_stream_t* stream = lua_newuserdata(L, sizeof(*stream));
+  mz_streamp miniz_stream = &(stream->stream);
+  luaL_getmetatable(L, "miniz_inflator");
+  lua_setmetatable(L, -2);
+  memset(miniz_stream, 0, sizeof(*miniz_stream));
+  int status = mz_inflateInit(miniz_stream);
+  if (status != MZ_OK) {
+    const char* msg = mz_error(status);
+    if (msg) {
+      luaL_error(L, "Problem initializing stream: %s", msg);
+    } else {
+      luaL_error(L, "Problem initializing stream");
+    }
+  }
+  stream->mode = 1;
+  return 1;
+}
+
+static int lmz_deflator_gc(lua_State* L) {
+  lmz_stream_t* stream = luaL_checkudata(L, 1, "miniz_deflator");
+  mz_deflateEnd(&(stream->stream));
+  return 0;
+}
+
+static int lmz_inflator_gc(lua_State* L) {
+  lmz_stream_t* stream = luaL_checkudata(L, 1, "miniz_inflator");
+  mz_inflateEnd(&(stream->stream));
+  return 0;
+}
+
+static const char* flush_types[] = {
```

Comment:
Why is this not an enum?

Reply:
Lua requires a string array for [luaL_checkopt](https://www.lua.org/manual/5.1/manual.html#luaL_checkoption), which is where this is used.

Reply:
You probably want to use optint to stay consistent with [deflate](https://github.com/FiniteReality/luvi/blob/1379d0db25194d723afbc730e7238ed11a4e45bd/src/lminiz.c#L346).

Reply:
Those functions appear to use an integer because it's a bitfield, rather than an enumeration. Personally, I think using a string is more favourable here but I will gladly change it.

Reply:
I guess either works.